src/main/java/com/google/cloud/genomics/dataflow/functions/CombineShardsFn.java

```diff
@@ -0,0 +1,189 @@
+package com.google.cloud.genomics.dataflow.functions;
+
+import com.google.api.services.storage.model.ComposeRequest;
+import com.google.api.services.storage.model.ComposeRequest.SourceObjects;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.Storage.Objects.Compose;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.util.GcsUtil;
+import com.google.cloud.dataflow.sdk.util.gcsfs.GcsPath;
+import com.google.cloud.dataflow.sdk.util.Transport;
+import com.google.cloud.dataflow.sdk.values.PCollectionView;
+import com.google.cloud.genomics.dataflow.readers.bam.BAMIO;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.ShardedBAMWritingOptions;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Lists;
+
+import htsjdk.samtools.BAMIndexer;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class CombineShardsFn extends DoFn<String, String> {
```

Comment:
Please add a comment here to give the overview of why someone might want to use this DoFn in their pipeline, thanks!